src/Ethereum/ABI/ParamFactory.cpp

```diff
@@ -74,9 +94,25 @@ std::shared_ptr<ParamBase> ParamFactory::make(const std::string& type) {
     return param;
 }
 
+std::string joinArrayElems(const std::vector<std::string>& strings) {
+    auto jvalues = std::vector<json>(strings.size());
```

Comment:
I think we can just `json::array()` instead of `vector<json>`?

Reply:
The quoting is still a problem, as the values are already represented as string, json will not convert simple values automatically.  I changed to json::array(), but the per-value parsing is still there.